lib/stages/frbNetworkProcess.cpp

```diff
@@ -265,7 +278,11 @@ void frbNetworkProcess::main_thread() {
         count++;
     }
     ping_cv.notify_all();
-    ping_thread.join();
+
+    if (send_ping_thread.joinable()) {
+        send_ping_thread.join();
+        receive_ping_thread.join();
```

Comment:
I think you should check if this thread is joinable as well.

Reply:
Done in 6ac68e8a0dd6345827afb077558e048b28cd6bfe